cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsDownscaleService.java

```diff
@@ -235,7 +236,12 @@ private List<String> terminateInstances(Long stackId, List<String> instanceIdsTo
                         .collect(Collectors.toList());
                 LOGGER.info("Running instances on AWS to terminate: {}", runningInstances);
                 if (!runningInstances.isEmpty()) {
-                    amazonEC2Client.terminateInstances(new TerminateInstancesRequest().withInstanceIds(runningInstances));
+                    if (runningInstances.size() < instanceIdsToDelete.size()) {
+                        return terminateInstances(stackId, runningInstances, amazonEC2Client);
```

Comment:
are you sure about introducing `recursion`? I think it's easier to read a loop, and in this case it wouldn't be too complicated to refactor this with the same behavior.

Reply:
Good point; I wrote a while loop instead with a safe logic